torch/csrc/cuda/Event.cpp

```diff
@@ -92,7 +89,8 @@ static PyObject * THCPEvent_get_cuda_event(THCPEvent *self) {
 
 static PyObject * THCPEvent_get_device(THCPEvent *self) {
   HANDLE_TH_ERRORS
-  return THPUtils_packInt64(self->cuda_event.device_index());
+  at::optional<at::Device> device = self->cuda_event.device();
+  return device ? THPDevice_New(device.value()) : Py_BuildValue((char*)"");
```

Comment:
@colesbury 

I am not sure if this is proper way to do it. Do we have a better way for returning an optional value?

Besides, this function throws an exception if the device arg is `None`. I think this behavior makes sense, as it prevents users from getting the device from a not-created event, and later pass it to `from_ipc_handle`.

Reply:
```c++
if (!device) {
  Py_RETURN_NONE;
}
return THPDevice_New(device.value());
```

In general, we map missing optional values to None in Python. For example, Tensor.grad and Tensor.grad_fn are None when the tensor doesn't have a gradient or doesn't require grad.